### PR TITLE
validate get-mapping-fields command

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/28266

## Screenshots
Error example:
image

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 1]
```diff
@@ -867,3 +868,23 @@ def is_there_a_runnable(self) -> bool:
             self.handle_error(error, code, file_path=self.file_path)
             return False
         return True
+
+    def is_valid_ismappble_field(self):
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:872]:
```suggestion
    def is_mapping_fields_command_exist(self) -> bool:
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 2]
```diff
@@ -867,3 +868,23 @@ def is_there_a_runnable(self) -> bool:
...
+        """
+        Check if get-mapping-fields command exists in the yml is {ismappble: True}
+        Returns:
+            True if get-mapping-fields commands exist in the yml, else False.
+
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:877]:
```suggestion
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 3]
```diff
@@ -867,3 +868,23 @@ def is_there_a_runnable(self) -> bool:
             self.handle_error(error, code, file_path=self.file_path)
             return False
         return True
+
+    def is_valid_ismappble_field(self):
+        """
+        Check if get-mapping-fields command exists in the yml is {ismappble: True}
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:874]:
```suggestion
        Check if get-mapping-fields command exists in the YML if  the ismappble field is set to true
```

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 4]
```diff
@@ -867,3 +868,23 @@ def is_there_a_runnable(self) -> bool:
...
+        Returns:
+            True if get-mapping-fields commands exist in the yml, else False.
+
+        """
+        script = self.current_file.get('script', {})
+        command_names = []
+        if script.get('ismappable'):
+            commands = script.get('commands', [])
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/integration.py:882]: you can do it in a one-liner and to be a set for better performance:
```python
command_names = {command['name'] for command in script.get('commands', [])}
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 5]
```diff
@@ -560,6 +563,27 @@ def test_are_tests_configured(self, file_path: str, file_type: str, expected: bo
         validator = IntegrationValidator(structure_validator)
         assert validator.are_tests_configured() == expected
 
+    GET_MAPPING_FIELDS = [
+        (VALID_INTEGRATION_IS_MAPPING_FIELDS, 'integration', True),
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:567]: avoid using files for test data, use the test suite instead

File: demisto_sdk/commands/common/hook_validations/integration.py [hunk 6]
```diff
@@ -867,3 +868,19 @@ def is_there_a_runnable(self) -> bool:
...
+            True if get-mapping-fields commands exist in the yml, else False.
+        """
+        script = self.current_file.get('script', {})
+        if script.get('ismappable'):
+            command_names = {command['name'] for command in script.get('commands', [])}
+            if 'get-mapping-fields' not in command_names:
+                self.is_valid = False
+                error, code = Errors.missing_get_mapping_fields_command()
+                self.handle_error(error, code, file_path=self.file_path)
+                return False
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/integration.py:885]:
```suggestion
                if self.handle_error(error, code, file_path=self.file_path):
                    self.is_valid = False
                    return False
```

File: demisto_sdk/commands/common/errors.py [hunk 7]
```diff
@@ -412,6 +413,11 @@ def parameter_missing_for_feed(name, correct_format):
                f'parameter "{name}" is missing or malformed in the YAML file.\n' \
                f'The correct format of the parameter should be as follows:\n{correct_format}'
 
+    @staticmethod
+    @error_code_decorator
+    def missing_get_mapping_fields_command():
+        return 'The command "get-mapping-fields" is missing from the yml file. Please add it.'
```
Comment (Itay4) [inline demisto_sdk/commands/common/errors.py:419]:
```suggestion
        return 'The command "get-mapping-fields" is missing from the YML file and is required as the ismappable field is set to true.'
```

**Review by Itay4 — APPROVED** [review]
please address @roysagi's comment before merging

**Comment by Shellyber** [comment]
@Itay4 I did

**Pull request merged**